.github/workflows/release.yml

```diff
@@ -1,24 +1,33 @@
 name: Release
 on:
-  release:
-    types:
-      - created
-
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The version to release (which will also be used for the tag)'
+        required: true
+        type: string
+permissions:
+  contents: write # To write the version change to the repository
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Setup Java
-        uses: actions/setup-java@v4
+      - name: Update version in gradle.properties
+        run: |
+          sed -i 's/^version=.*$/version=$VERSION/' gradle.properties
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git add gradle.properties
+          git commit -m "Update version to $VERSION"
+          git tag -a $VERSION -m "Release $VERSION"
+          git push --tags
         with:
-          distribution: 'temurin'
-          java-version: 11
-      - name: Setup Gradle
-        uses: gradle/actions/setup-gradle@v4
-      - name: Publish with Gradle
-        run: ./gradlew publish
-        with:
-          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
-          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+          VERSION: ${{ inputs.version }}
+      - name: Create release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          version: ${{ inputs.version }}
+        run: |
+          gh release create "$tag" --generate-notes
```

Comment:
```suggestion
          gh release create "$tag" --generate-notes --draft
```

Reply:
Why would you create a draft release?

Reply:
To not notify people until the publishing is actually done :)

Reply:
I now publish before creating the release, so we can notify people?